store/redis.go

```diff
@@ -0,0 +1,96 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// RedisConfig ... user configurable
+type RedisConfig struct {
+	Endpoint string
+	Password string
+	DB       int
+	Eviction time.Duration
+	Profile  bool
+}
+
+// RedStore ... Redis storage backend implementation
+type RedStore struct {
+	eviction time.Duration
```

Comment:
do we need concurency handling

Reply:
Concurrency handling is done by the caller

Reply:
which caller are we talking about here? Do you mean that calls to the RedStore are always linearized b/c they are made by Put/GetHandlers and those take a mutex? Even if this, I don't see that this is a good idea and I would agree with Bowen here I think, unless I'm missing some context on how exactly this is used.

Reply:
@samlaf the caller holds a mutex when referencing cache or fallback dependencies: https://github.com/Layr-Labs/eigenda-proxy/blob/main/store/router.go#L20-L24

Reply:
Why don't we put the mutex in this struct instead? Is it because we need to hold the same mutex to interact with multiple different cache? Can we add a comment above RedStore saying that this is not goroutine safe?

Reply:
can definitely update with a comment!

Reply:
[d10f376](https://github.com/Layr-Labs/eigenda-proxy/pull/111/commits/d10f37634ba625d03c4178763d49a374679fe6bb)